### PR TITLE
MAUI Extension Continued

### PR DESCRIPTION
**Pull request by Sweekriti91**
Apologies in Advance, most of the files changes are the Test samples for MAUI end to end scenario, feel free to skip those 😅

Fixes Issues :

https://github.com/dotnet/upgrade-assistant/issues/622
https://github.com/dotnet/upgrade-assistant/issues/586

- Adds custom step to adjust TFM for `net6.0-ios` and `net6.0-android` based on Xamarin Platform Source project
- Adds TargetSelectorFilter for `net6.0-ios` and `net6.0-android` targets for MAUI Extension
- Adds ComponentIdentifier for MAUI exntesion
- End to End Tests Added for iOS and Droid
- Adds Package Maps for packages to be removed from old Xamarin Projects

### CODE REVIEW

File: src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/ProjectComponents.cs
Comment (twsouthwick) [inline src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/ProjectComponents.cs:19]: Do we want to add a Maui component as well?

File: src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiPlatformTargetFrameworkUpgradeStep.cs
Comment (twsouthwick) [inline src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiPlatformTargetFrameworkUpgradeStep.cs:80]: Maybe return a failed initialize result with the message instead of throwing up

File: src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiPlatformTargetFrameworkUpgradeStep.cs
Comment (twsouthwick) [inline src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiPlatformTargetFrameworkUpgradeStep.cs:84]: I think it would be clearer to parse the component flag property as an enum and compare that here

File: src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiPlatformTargetFrameworkUpgradeStep.cs
Comment (twsouthwick) [inline src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiPlatformTargetFrameworkUpgradeStep.cs:77]: Looks like this stuff is duplicated. Could this be refactored to have a `GetExpectedTfm`?

I also wonder if it would work to subclass the SetTfmStep for this....

File: src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/XamarinFormsCheck.cs
Comment (twsouthwick) [inline src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/XamarinFormsCheck.cs:34]: There's an `SR.Format` method that could be used here to ensure everything is using invariant culture (feel free to add more overloads to that method). That method is shared among all projects.

File: src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/XamarinFormsCheck.cs
Comment (twsouthwick) [inline src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/XamarinFormsCheck.cs:69]: This feels really fragile.... there's an `IVersionComparer` that may be useful - feel free to add more methods for things like parse or something.

**Review by twsouthwick — APPROVED** [review]
A few comments inline (hopefully no annoying "nits" :)) but overall LGTM

**Comment by twsouthwick** [comment]
Oh - it would be great to have some unit tests as well for the selector/filter/etc

File: src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/ProjectComponents.cs
Comment (Sweekriti91) [reply src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/ProjectComponents.cs:19]: I figured I'd add them incrementally as I add the next steps. next PR will include the MAUI components + checks for them so that it adds the correct templates. I guess I could add them now but it won't be used anywhere at the moment (relative to this PR at least).

File: src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/XamarinFormsCheck.cs
Comment (Sweekriti91) [reply src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/XamarinFormsCheck.cs:34]: Hmm I saw the other Checks followed this format, so I stuck with it. `SR.Format` is damn cool though, should all the check messages be updated to this format?

File: src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/ProjectComponents.cs
Comment (Sweekriti91) [reply src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/ProjectComponents.cs:19]: @twsouthwick wanted to check if the reason makes sense to you? made all the other suggested updates! 😄

File: src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiPlatformTargetFrameworkUpgradeStep.cs
Comment (Sweekriti91) [reply src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiPlatformTargetFrameworkUpgradeStep.cs:74]: @twsouthwick I couldn't find an easy way to subclass the step from the extension project. I did create a `GetExpectedTfm` and refactored the code. Let me if you still want to try something here?

**Comment by Sweekriti91** [comment]
Oh hmm dunno why AzDo build checks aren't coming through. @twsouthwick  I did add the MauiFilter tests and the builds are passing now! 😄  Just waiting on your reviews on the last 2 comments before I merge this!

File: src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/ProjectComponents.cs
Comment (twsouthwick) [reply src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/ProjectComponents.cs:19]: That's fine with me!

**Comment by twsouthwick** [comment]
LGTM!

**Pull request merged**
